src/databricksbundle/notebook/GithubLinkGenerator.py

```diff
@@ -0,0 +1,89 @@
+import tomlkit
+import traceback
+
+from inspect import getsource
+from pathlib import Path
+
+
+class GithubLinkGenerator:
+    def __get_repo_name(self, module):
+        special_repos = {"daipecore": "daipe-core", "featurestorebundle": "feature-store-bundle"}
+
+        if module in special_repos.keys():
+            return special_repos[module]
+
+        return self.__get_repo_name_with_bundle(module)
+
+    def __get_repo_name_with_bundle(self, module):
+        index_of_bundle = module.find("bundle")
+
+        return module[:index_of_bundle] + "-" + module[index_of_bundle:]
+
+    def __get_module_version(self, module_name):
+        base_path = Path("/".join(list(Path.cwd().parts[0:5])))
+        lockfile_path = base_path.joinpath("poetry.lock")
+
+        with lockfile_path.open("r") as f:
+            config = tomlkit.parse(f.read())
+
+        for module in config["package"]:
+            if module["name"] == module_name:
+                return module["version"]
+
+        raise Exception("Specified module could not be found, make sure the the module is imported and developed by DAIPE")
+
+    def __get_display_html(self):
+        import IPython
+
+        ipython = IPython.get_ipython()
+
+        if not hasattr(ipython, "user_ns") or "displayHTML" not in ipython.user_ns:
+            raise Exception("displayHTML cannot be resolved")
+
+        return ipython.user_ns["displayHTML"]
+
+    def __get_github_url_from_module(self, module):
+        module_path = module.__module__
+        module_name_parent = module_path.split(".")[0]
+        github_repo_name = self.__get_repo_name(module_name_parent)
+        module_file_path = module_path.replace(".", "/") + ".py"
+        version = self.__get_module_version(github_repo_name)
+
+        return self.__generate__github_url(github_repo_name, version, module_file_path)
+
+    def __generate__github_url(self, repo_name: str, version: str, file_path: str, lineno: str = ""):
```

Comment:
`lineno: str = ""` mutable default arguments are bad
`lineno: str = None`
`lineno = "" if lineno is not None else lineno`